File: frontend/src/app/services/api.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment'; // <- путь от services
import { Note } from '../models/note.model';
import { User } from '../models/user.model';

@Injectable({
  providedIn: 'root'
})
export class ApiService {
  //private baseUrl = 'https://localhost:7105/api';
  private baseUrl = environment.apiUrl;

  constructor(private http: HttpClient) {}

  //  // Регистрация
  // register(user: any): Observable<any> {
  //   return this.http.post(`${this.baseUrl}/Users/register`, user);
  // }

  // // Авторизация
  // login(user: any): Observable<any> {
  //   return this.http.post(`${this.baseUrl}/Users/login`, user);
  // }

  // // Получение заметок по ID пользователя
  // getNotes(userId: number): Observable<any> {
  //   return this.http.get(`${this.baseUrl}/Notes/user/${userId}`);
  // }

  // // Добавление новой заметки
  // addNote(note: any): Observable<any> {
  //   return this.http.post(`${this.baseUrl}/Notes`, note);
  // }

  // // Обновление заметки
  // updateNote(id: number, note: any): Observable<any> {
  //   return this.http.put(`${this.baseUrl}/Notes/${id}`, note);
  // }
  
  // // Удаление заметки
  // deleteNote(id: number): Observable<any> {
  //   return this.http.delete(`${this.baseUrl}/Notes/${id}`);
  // }

  
  
  // Регистрация
  register(user: User): Observable<User> {
    return this.http.post<User>(`${this.baseUrl}/Users/register`, user);
  }

  // Авторизация
  login(payload: { email: string; password: string }): Observable<User> {
    return this.http.post<User>(`${this.baseUrl}/Users/login`, payload);
  }

  // Получение заметок по ID пользователя
  getNotes(userId: number): Observable<Note[]> {
    return this.http.get<Note[]>(`${this.baseUrl}/Notes/user/${userId}`);
  }

  // Добавление новой заметки
  addNote(note: Note): Observable<Note> {
    return this.http.post<Note>(`${this.baseUrl}/Notes`, note);
  }

  // Обновление заметки
  updateNote(id: number, note: Note): Observable<Note> {
    return this.http.put<Note>(`${this.baseUrl}/Notes/${id}`, note);
  }

  // Удаление заметки — backend у тебя возвращает список заметок -> Observable<Note[]>
  deleteNote(id: number): Observable<Note[]> {
    return this.http.delete<Note[]>(`${this.baseUrl}/Notes/${id}`);
  }
}

File: windows/CalendarApp.Desktop/Services/ApiService.cs
// Services/ApiService.cs
using System.Net.Http;
using System.Net.Http.Json;
using CalendarApp.Desktop.Models;
using CalendarAppWinForms.Models;
using Newtonsoft.Json;

namespace CalendarApp.Desktop.Services
{
    public class ApiService
    {
        private readonly HttpClient _http;

        public ApiService(string baseUrl)
        {
            _http = new HttpClient { BaseAddress = new Uri(baseUrl) };
        }

        public async Task<UserDto?> LoginAsync(string email, string password)
        {
            //var payload = new LoginRequestDto { Email = email, PasswordHash = password };
            var payload = new LoginRequestDto { Email = email, Password = password };
            var res = await _http.PostAsJsonAsync("api/users/login", payload);
            if (!res.IsSuccessStatusCode) return null;
            return await res.Content.ReadFromJsonAsync<UserDto>();
        }

        public async Task<(bool Success, string? ErrorMessage)> RegisterAsync(RegisterRequest request)
        {
            var res = await _http.PostAsJsonAsync("api/users/registeruser", request);
            //var res = await _http.PostAsJsonAsync("api/Users/register", request);
            if (res.IsSuccessStatusCode)
                return (true, null);

            var error = await res.Content.ReadAsStringAsync();
            return (false, error);
        }
        //        public async Task<(bool Success, string? ErrorMessage)> RegisterAsync(RegisterRequest request)
        //{
        //    var res = await _http.PostAsJsonAsync("Users/registeruser", request);
        //    if (res.IsSuccessStatusCode)
        //        return (true, null);

        //    var error = await res.Content.ReadAsStringAsync();
        //    return (false, error);
        //}


        public async Task<List<NoteDto>> GetNotesForUserAsync(int userId)
        {
            var res = await _http.GetAsync($"api/notes/user/{userId}");
            res.EnsureSuccessStatusCode();
            return await res.Content.ReadFromJsonAsync<List<NoteDto>>() ?? new List<NoteDto>();
        }

        public async Task<NoteDto?> CreateNoteAsync(NoteDto note)
        {
            var res = await _http.PostAsJsonAsync("api/notes", note);
            if (!res.IsSuccessStatusCode) return null;
            return await res.Content.ReadFromJsonAsync<NoteDto>();
        }

        public async Task<bool> DeleteNoteAsync(int id)
        {
            try
            {
                var response = await _http.DeleteAsync($"api/notes/{id}");
                return response.IsSuccessStatusCode;
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Ошибка при удалении заметки: {ex.Message}");
                return false;
            }
            //var res = await _http.DeleteAsync($"api/Notes/{id}");
            //return res.IsSuccessStatusCode;
        }

        public async Task<bool> UpdateNoteAsync(int id, NoteDto note)
        {
            var res = await _http.PutAsJsonAsync($"api/notes/{id}", note);
            return res.IsSuccessStatusCode;
        }
        // UploadImageAsync при необходимости



        public async Task<UserDto?> GetUserByIdAsync(int id)
        {
            var response = await _http.GetAsync($"api/users/{id}");
            if (!response.IsSuccessStatusCode)
                return null;

            var json = await response.Content.ReadAsStringAsync();
            return JsonConvert.DeserializeObject<UserDto>(json);
        }
    }
}
